neural_tangents/stax.py

```diff
@@ -986,6 +986,77 @@ def kernel_fn(kernels):
   return init_fn, apply_fn, kernel_fn
 
 
+
+@_layer
+def FanInConcat(axis=-1):
+  """Layer construction function for a fan-in sum layer.
+
+  Based on `jax.experimental.stax.FanInConcat`.
+  """
+  init_fn, apply_fn = ostax.FanInConcat(axis=axis)
+  def kernel_fn(kernels):
+    is_gaussian = all(ker.is_gaussian for ker in kernels)
+
+    if not is_gaussian:
+      raise NotImplementedError('`FanInConcat` layer is only implemented for the '
+                                'case if all input layers guaranteed to be mean'
+                                '-zero gaussian, i.e. having all `is_gaussian'
+                                'set to `True`.')
+
+    marginal, cross = kernels[0].marginal, kernels[0].cross
+    shape1, shape2 = kernels[0].shape1, kernels[0].shape2
+    if not all(k.marginal == marginal and
+               k.cross == cross
+               for k in kernels):
+      raise NotImplementedError('`FanInConcat` layer is only implemented for the '
+                                'case if all input layers output the same type'
+                                'of covariance matrices, i.e. having all '
+                                'matching `marginal` and `cross` attributes')
+
+    # If kernels have different height/width order, transpose some of them.
+    n_kernels = len(kernels)
+    n_height_width = sum(ker.is_height_width for ker in kernels)
+
+    if n_height_width == n_kernels:
+      is_height_width = True
+
+    elif n_height_width >= n_kernels / 2:
+      is_height_width = True
+      for i in range(n_kernels):
+        if not kernels[i].is_height_width:
+          kernels[i] = _flip_height_width(kernels[i])
+
+    else:
+      is_height_width = False
+      for i in range(n_kernels):
+        if kernels[i].is_height_width:
+          kernels[i] = _flip_height_width(kernels[i])
+
+    if not all([k.shape1 == shape1 and k.shape2 == shape2 for k in kernels]):
+      raise ValueError('All shapes should be equal in FanInConcat.')
+
+    x1_is_x2 = kernels[0].x1_is_x2
+    kers = []
+    for i in range(4):
+      if all(ker[i] is None for ker in kernels):
+        kers.append(None)
+      else:
+        #kers.append(np.concatenate( [ker[i] for ker in kernels] ,axis=axis))
+        kers.append( [ker[i] for ker in kernels]  )
+    kers = tuple(kers)
+    #kers = tuple(None if all(ker[i] is None for ker in kernels) else
+    #             sum(ker[i] for ker in kernels) for i in range(4))
+    is_input = kernels[0].is_input
+    return Kernel(*(
+        kers + (is_gaussian, is_height_width, marginal, cross, None, None,
+                x1_is_x2, is_input)))
+
+  return init_fn, apply_fn, kernel_fn
+
+
+
+
+
```

Comment:
Since this isn't working currently, please remove this, so that this block doesn't end up merged into the codebase. FYI, we're planning to add the `FanInConcat` layer soon!